dynamo/tools/clustering.py

```diff
@@ -227,31 +251,62 @@ def leiden(
 
 
 def louvain(
-    adata,
-    resolution=1.0,
-    use_weight=True,
-    adj_matrix=None,
-    adj_matrix_key=None,
-    randomize=False,
-    result_key=None,
-    layer=None,
-    obsm_key=None,
-    selected_cluster_subset: list = None,
-    selected_cell_subset=None,
-    directed=False,
-    copy=False,
+    adata: AnnData,
+    resolution: float = 1.0,
+    use_weight: bool = True,
+    adj_matrix: Optional[csr_matrix] = None,
+    adj_matrix_key: Optional[str] = None,
+    randomize: Optional[int] = None,
+    result_key: Optional[str] = None,
+    layer: Optional[str] = None,
+    obsm_key: Optional[str] = None,
+    selected_cluster_subset: Optional[list] = None,
+    selected_cell_subset: Optional[list] = None,
+    directed: bool = False,
+    copy: bool = False,
     **kwargs
 ) -> anndata.AnnData:
-    """Louvain implementation from cdlib.
-    For other community detection general parameters, please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
-    "Louvain maximizes a modularity score for each community. The algorithm optimises the modularity in two elementary phases: (1) local moving of nodes; (2) aggregation of the network. In the local moving phase, individual nodes are moved to the community that yields the largest increase in the quality function. In the aggregation phase, an aggregate network is created based on the partition obtained in the local moving phase. Each community in this partition becomes a node in the aggregate network. The two phases are repeated until the quality function cannot be increased further." - cdlib
+    """Apply louvain clustering to adata.
+    For other community detection general parameters,
+    please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
+    "Louvain maximizes a modularity score for each community.
+    The algorithm optimises the modularity in two elementary phases:
+    (1) local moving of nodes;
+    (2) aggregation of the network.
+    In the local moving phase,
+    individual nodes are moved to the community that yields the largest increase in the quality function.
+    In the aggregation phase,
+    an aggregate network is created based on the partition obtained in the local moving phase.
+    Each community in this partition becomes a node in the aggregate network.
+    The two phases are repeated until the quality function cannot be increased further." - cdlib
 
-    Parameters
-    ----------
-    resolution : float, optional
-        change the size of the communities, default to 1.
-    randomize : bool, optional
-        "randomState instance or None, optional (default=None). If int, random_state is the seed used by the random number generator; If RandomState instance, random_state is the random number generator; If None, the random number generator is the RandomState instance used by np.random." - cdlib
+    Args:
+        adata: an adata object
+        resolution: determines clustering level of detail.
```

Comment:
don't be lazy, use complete sentence. `The resolution parameter that determines clustering level of detail.`